hooks/usePoints.ts

```diff
@@ -0,0 +1,21 @@
+import useUser from '@/hooks/useUser';
+import { usePointsStore } from '@/store/usePointsStore';
+import { useEffect } from 'react';
+
+export const usePoints = () => {
+  const { user } = useUser();
+  const { points, isLoading, error, fetchPoints } = usePointsStore();
+
+  useEffect(() => {
+    if (user && points.userRewardsSummary.totalPoints === 0) {
+      fetchPoints();
+    }
+  }, [user, points.userRewardsSummary.totalPoints, fetchPoints]);
```

Comment:
@MayankMittal1 when fetchPoints() updates `points.userRewardsSummary.totalPoints`, it could trigger an infinite loop